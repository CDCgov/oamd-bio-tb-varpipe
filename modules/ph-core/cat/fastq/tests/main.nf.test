nextflow_process {

    name "Test Process CAT_FASTQ"
    script "../main.nf"
    process "CAT_FASTQ"

    tag "modules"
    tag "modules_phcore"
    tag "cat"
    tag "cat/fastq"

    test("test_cat_fastq_single_end") {

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_paired_end") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test2_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test2_2.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_single_end_same_name") {

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_paired_end_same_name") {

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_single_end_single_file") {

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_single_end - stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_paired_end - stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test2_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test2_2.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_single_end_same_name - stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_paired_end_same_name - stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_fastq_single_end_single_file - stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
