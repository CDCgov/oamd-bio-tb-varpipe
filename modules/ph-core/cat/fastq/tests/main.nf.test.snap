{
    "test_cat_fastq_single_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,ee314a9bd568d06617171b0c85f508da"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,ee314a9bd568d06617171b0c85f508da"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:32:50.149369"
    },
    "test_cat_fastq_single_end_same_name": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,3ad9406595fafec8172368f9cd0b6a22"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,3ad9406595fafec8172368f9cd0b6a22"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:33:01.666427"
    },
    "test_cat_fastq_single_end_single_file": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:33:12.457508"
    },
    "test_cat_fastq_paired_end_same_name": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.merged.fastq.gz:md5,3ad9406595fafec8172368f9cd0b6a22",
                            "test_2.merged.fastq.gz:md5,a52cab0b840c7178b0ea83df1fdbe8d5"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.merged.fastq.gz:md5,3ad9406595fafec8172368f9cd0b6a22",
                            "test_2.merged.fastq.gz:md5,a52cab0b840c7178b0ea83df1fdbe8d5"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:33:07.619561"
    },
    "test_cat_fastq_single_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:33:17.914863"
    },
    "test_cat_fastq_paired_end_same_name - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test_2.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test_2.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:33:35.658913"
    },
    "test_cat_fastq_single_end_same_name - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:33:29.691018"
    },
    "test_cat_fastq_paired_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.merged.fastq.gz:md5,3ad9406595fafec8172368f9cd0b6a22",
                            "test_2.merged.fastq.gz:md5,a52cab0b840c7178b0ea83df1fdbe8d5"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.merged.fastq.gz:md5,3ad9406595fafec8172368f9cd0b6a22",
                            "test_2.merged.fastq.gz:md5,a52cab0b840c7178b0ea83df1fdbe8d5"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:32:56.53097"
    },
    "test_cat_fastq_paired_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test_2.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test_2.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-24T18:46:40.306477641"
    },
    "test_cat_fastq_single_end_single_file - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd4179ac1816befccf45d1c6584751d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-22T13:33:40.408276"
    }
}
